File: convalida-validators/src/test/java/convalida/validators/NotEmptyValidatorTest.java
package convalida.validators;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.when;

/**
 * @author Wellington Costa on 01/11/2017.
 */
public class NotEmptyValidatorTest extends BaseTest {

    @Test public void emptyValue() {
        NotEmptyValidator validator = new NotEmptyValidator(mockEditText, errorMessage, true);
        when(mockEditText.getText().toString()).thenReturn("");
        assertEquals(validator.validate(), false);
    }

    @Test public void validValue() {
        NotEmptyValidator validator = new NotEmptyValidator(mockEditText, errorMessage, true);
        when(mockEditText.getText().toString()).thenReturn("test");
        assertEquals(validator.validate(), true);
    }

}

File: convalida-validators/src/test/java/convalida/validators/NotEmptyValidatorTest.kt
package convalida.validators

import org.junit.Test

import org.junit.Assert.assertEquals
import org.mockito.Mockito.`when`

/**
 * @author Wellington Costa on 01/11/2017.
 */
class NotEmptyValidatorTest : BaseTest() {

    @Test
    fun emptyValue() {
        val validatorWithEditText = NotEmptyValidator(mockEditText, errorMessage)
        `when`(mockEditText.text.toString()).thenReturn("")
        assertEquals(validatorWithEditText.validate(), false)
    }

    @Test
    fun validValue() {
        val validatorWithEditText = NotEmptyValidator(mockEditText, errorMessage)
        `when`(mockEditText.text.toString()).thenReturn("test")
        assertEquals(validatorWithEditText.validate(), true)
    }

}